not-so-smart-contracts/substrate/weights_and_fees/README.md

```diff
@@ -0,0 +1,49 @@
+# Weights and Fees
+
+Weights and transaction fees are the two main ways to regulate the consumption of blockchain resources. The overuse of blockchain resources can allow a malicious actor to spam the network to cause a denial-of-service (DoS). Weights are used to manage the time it takes to validate the block. The larger the weight, the more "resources" / time the computation takes. Transaction fees provide an economic incentive to limit the number of resources used to perform operations; the fee for a given transaction is a function of the weight required by the transaction. 
+
+Weights can be fixed or a custom "weight annotation / function" can be implemented. A weight function can calculate the weight, for example, based on the number of database read / writes and the size of the input paramaters (e.g. a long `Vec<>`). To optimize the weight such that users do not pay too little or too much for a transaction, benchmarking can be used to empirically determine the correct weight in worst case scenarios.
```

Comment:
Similar to the other issue, we explain how it works, not what the issue is

Reply:
https://github.com/crytic/building-secure-contracts/pull/124/commits/a4fc1bd39a6dd24e5a305d1287e8e9c95860e192